gprofiler/profilers/node.py

```diff
@@ -180,13 +174,17 @@ def _validate_pid(expected_pid: int, sock: WebSocket) -> None:
     assert expected_pid == actual_pid, f"Wrong pid, expected {expected_pid}, actual {actual_pid}"
 
 
+@contextmanager
 def create_debugger_socket(nspid: int, ns_link_name: str) -> WebSocket:
-    debugger_url = _get_debugger_url()
-    sock = create_connection(url=debugger_url, timeout=15.0)
-    sock.settimeout(10)
-    _validate_ns_node(sock, ns_link_name)
-    _validate_pid(nspid, sock)
-    return sock
+    try:
+        debugger_url = _get_debugger_url()
+        sock = create_connection(url=debugger_url, timeout=15.0)
+        sock.settimeout(10)
+        _validate_ns_node(sock, ns_link_name)
+        _validate_pid(nspid, sock)
+        yield sock
+    finally:
+        _close_debugger(sock)
```

Comment:
If `_get_debugger_url` raised - this code runs and access the unbound `sock`. In such cases you need to set it to `None` (preferably outside of the `try` block) and in the `finally` you check `if sock is not None`.